grails-app/services/org/pih/warehouse/inventory/InventorySnapshotService.groovy

```diff
@@ -525,14 +525,15 @@ class InventorySnapshotService {
             row.put("Opening", balanceOpening)
             def includeRow = balanceOpening || balanceClosing || quantityAdjustments
             transactionTypeNames.each { String transactionTypeName ->
+                String translatedTransactionTypeName = transactionTypeName
                 if (transactionTypeName.contains(Constants.LOCALIZED_STRING_SEPARATOR)) {
-                    transactionTypeName = LocalizationUtil.getLocalizedString(transactionTypeName)
+                    translatedTransactionTypeName = LocalizationUtil.getLocalizedString(transactionTypeName)
                 }
                 def quantity =
                         transactionData.find {
-                            it.productCode == product.productCode && it.transactionTypeName == transactionTypeName
+                            it.productCode == product.productCode && it.transactionTypeName == transactionTypeName.split("\\|")[0]
```

Comment:
Why is this necessary? Shouldn't getLocalizedString be doing that for us?

Reply:
It does, but it compares `English == French` which always results in `false`. 
The `transactionData` in line `468` stores the `transactionTypeName` as English only, so that's why I made it

Reply:
I'll send some images from debugging in a moment to show what I mean

Reply:
So, the `transactionData` example object looks like this:
![Transaction data](https://i.imgur.com/cGY1x8b.png)

and the `transationTypeName` example looks like this:
![Transaction data](https://i.imgur.com/wrmrQf2.png)

so it was comparing `Consumption == Consommantion` for example, instead of comparing `Consumption == Consumption`

Reply:
Ok got it. I am going to write a comment in the ticket to explain what I'd like to see in the long term. But for now, you have the right approach. Let's just move that logic to a convenience method on the TransactionType class if possible.

@awalkowiak Are we allowed to do something like this
```
Boolean compareTo(String transactionTypeName)
Boolean equals(String transactionTypeName)
```
or would it be better not to implement methods that are defined by common interfaces?

Reply:
@jmiranda I think I'd go with something like `Boolean compareTransactionType(String transactionTypeName)` because the string that might be passed not always will be the `transactionTypeName`, so for clarity, I would avoid using `compareTo` and `equals` and strictly let others know what I want to compare.
cc @kchelstowski